src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -174,15 +199,21 @@ public void RemoveAt(int index)
         {
             get
             {
-                if (index >= _size)
+                // Following trick can reduce the range check by one comparison
+                if ((uint)index >= (uint)Count)
```

Comment:
Isn't this changing what exception gets thrown for negative values?  Negative values would have previously thrown an IndexOutOfRangeException, now they're going to throw an ArgumentOutOfRangeException.  While it may have been better to be consistent when the type was originally introduced, there doesn't seem to be a compelling reason to take such a breaking change now.